src/ExperimentalInternetComputer.mo

```diff
@@ -16,4 +16,18 @@ module {
   public let call : (canister : Principal, name : Text, data : Blob) ->
      async (reply : Blob) = Prim.call_raw;
 
+  /// Given an IC performance counter, `counter`, and computation, `comp`,
+  /// measures the cost of performing computation `comp()`.
+  /// Returns the difference between the state of `counter` before and after executing `comp()`.
+  /// NB: Currently, the only available counter is `0`,
+  /// measuring executed wasm instructions.
+  /// (see [Performance Counter](https://internetcomputer.org/docs/current/references/ic-interface-spec#system-api-performance-counter))
+  public func measureCounter(counter : Nat32) : (comp : () -> ()) -> Nat64 {
```

Comment:
I think we can hardcode 0 here, and change the name to `measureCycleCounter`? If we ever get another counter type, we probably want to give it a more meaningful name anyway, instead of relying on  a number type.

Reply:
https://internetcomputer.org/docs/current/references/ic-interface-spec#system-api-performance-counter

Reply:
It's instructions indeed.